execute/row_fn.go

```diff
@@ -17,6 +18,15 @@ type dynamicFn struct {
 	scope      compiler.Scope
 	fn         *semantic.FunctionExpression
 	recordName string
+	compiledFn *compiledFn
```

Comment:
Is my reading of the code correct that we have a cache of size 1? Basically in the case where the type is always the same we do not have to recomiple, but in cases where it changes we recompile each time it changes. So worst case is that we recompile everytime (which is what we do now anyway) and best case is we only compile once. LGTM